src/backlight/portfolio/portfolio.py

```diff
@@ -80,16 +80,55 @@ def construct_portfolio(
     for (trade, lot) in zip(trades, lot_size):
         trade["amount"] *= lot
 
+    new_mkt = [symbols2mkt.get(t.symbol) for t in trades]
+    assert len(principal) == len(trades)
+    assert len(lot_size) == len(trades)
+
     # Construct positions and return Portfolio
     positions = Parallel(n_jobs=-1, max_nbytes=None)(
         [
             delayed(calc_positions)(trade, market, principal=principal_per_asset)
-            for (trade, market, principal_per_asset) in zip(trades, mkt, principal)
+            for (trade, market, principal_per_asset) in zip(trades, new_mkt, principal)
         ]
     )
+
+    symbols = [p.symbol for p in positions]
+    if len(set(symbols)) != len(symbols):
+        positions = fusion_positions(positions)
+
     return Portfolio(positions)
 
 
+def fusion_positions(positions: List[Positions]) -> List[Positions]:
+    """
+    Take a list of Positions and sum those with the same symbols
+    args :
+        - positions : a list of Positions to fusion
+    """
+
+    unique_positions = []
+    symbols = [p.symbol for p in positions]
+    for position in positions:
```

Comment:
Is it better to use following `for` loop?

```
for symbol in set(symbols):
```

You need not to write following lines:

```
for _ in range(symbols.count(s)):
                symbols.remove(s)
```